runtime/src/platform_utils.cpp

```diff
@@ -0,0 +1,49 @@
+#include "platform_utils.hpp"
+
+#if __linux__
+
+int sched_getcpu() {
+    return ::sched_getcpu();
+}
+
+#elif __APPLE__
+
+int sched_getcpu() {
+    int cpu = -1;
+    size_t len = sizeof(cpu);
+
+    if (sysctlbyname("hw.cpulocation", &cpu, &len, NULL, 0) == -1) {
+        return -1;  // Handle error
+    }
+    return cpu;
+}
+
+int sched_getaffinity(pid_t pid, size_t cpusetsize, cpu_set_t *mask) {
+    // Simulate affinity as macOS does not support setting/getting affinity
```

Comment:
Can't we emulate it with `thread_policy_set`?

Reply:
i think should be done with something like that, right now it is just a stub, will explore.

Reply:
will create a new issue for this one, currently this is added as just a no operation stub.